DOCS-8624-Scripting-index-edits-kt (#930)
mulesoft/docs-connectors: scripting/2.0/modules/ROOT/pages/edits/index.adoc

= Scripting Module 2.0 - Mule 4
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:page-aliases: connectors::scripting/scripting-module.adoc

Support Category: https://www.mulesoft.com/legal/versioning-back-support-policy#anypoint-connectors[Select]

> The Scripting Module executes custom logic written in a scripting language. In some cases, you might need to create custom code to perform all or part of a complex task, or to reuse modules that you have already written.

Reviewer: ```suggestion
Scripting Module executes custom logic written in a scripting language. In some cases, you might need to create custom code to perform all or part of a complex task, or to reuse modules that you have already written.
```

> Since version 2.0 you must provide a compliant `jsr-223` scripting language engine.

Reviewer: ```suggestion
Starting with Scripting Module version 2.0, you must provide a compliant `jsr-223` scripting language engine.
```

> For information about compatibility and fixed issues, see xref:release-notes::mule-runtime/module-scripting.adoc[Scripting Module Release Notes]

Reviewer: ```suggestion
For information about compatibility and fixed issues, see Scripting Module Release Notes.
```
Move link to See Also.

== Prerequisites

> To use this module, you must be familiar with the following concepts:

Reviewer: ```suggestion
To use this module, you must be familiar with:
```

* Anypoint Connectors
* Mule runtime engine (Mule)
* Elements in a Mule flow
* Global elements

> * Understand how to create a Mule app using Design Center or Anypoint Studio

Reviewer: ```suggestion
* Creating a Mule app using Design Center or Anypoint Studio
```

Before creating an app, you must have access to Anypoint Platform.

> == Common Use Cases For the Module

Reviewer: ```suggestion
== Common Use Cases for the Module
```

> The Scripting Module performs logic in a supported language, which enables you to access variables and the registry, log information via the Mule logger, read payloads, and so on.

Reviewer: ```suggestion
Scripting Module performs logic in a supported language, which enables you to access variables and the registry, log information via the Mule logger, read payloads, and so on.
```

> For example use cases, see xref:scripting-module-examples.adoc[Examples].

Reviewer: ```suggestion
For example use cases, see Examples.
```
Move link to See Also.
